core/src/main/scala/kafka/server/AbstractFetcherThread.scala

```diff
@@ -461,8 +510,9 @@ abstract class AbstractFetcherThread(name: String,
         val maybeTruncationComplete = fetchOffsets.get(topicPartition) match {
           case Some(offsetTruncationState) =>
             val state = if (offsetTruncationState.truncationCompleted) Fetching else Truncating
+            // Resetting `lastFetchedEpoch` since we are truncating and don't expect diverging epoch in the next fetch
```

Comment:
This is a little unclear to me. I guess it is safe to reset `lastFetchedEpoch` as long as we reinitialize it after the next leader change. On the other hand, it seems safer to always retain the value.

Reply:
Updated.